### DISCUSSION TITLE
OneDrive results not appearing in Search Results #1607

### DISCUSSION THREAD

**Discussion by Naveen-Rangarajan** [microsoft-search/pnp-modern-search#1668] [General]
In continuation to #1607, opening this issue here for discussion.

As Suggested by @wobba, The OneDrive items are not displayed in the Debug mode under Available layouts. How to check if this issue is related to API?

**Screen shot 1** below with some random keyword and Query Modification
![image](https://user-images.githubusercontent.com/61411651/147071091-3d7389bd-cf08-42f3-81fa-0b684372d89a.png)

**Screen shot 2** below with keyword from OneDrive and Query Modification
![image](https://user-images.githubusercontent.com/61411651/147071675-41e7f05a-5483-4977-bd06-b543c53011f9.png)

**Comment by FranckyC** [comment]
Hi @Naveen-Rangarajan, did you try the latest 4.5.3 version? We removed the 'Include OneDrive results' toggle option as they are now included in results by default.

**Reply by Naveen-Rangarajan** [reply]
The version we are using is 3.22 where 'Include OneDrive results' toggle option has already been removed. The issue is not here rather on the API or not sure how to check if the API has been blocked.

ODB items are retrieved in my Dev Tenant and not in my Client's Test and PROD tenant. All configs are same, able to access both the below links -

https://tenant-my.sharepoint.com/ 
path:https://tenant.sharepoint.com/

Still not sure where the underlying issue is.

**Reply by FranckyC** [reply]
Do you have any error in the browser network tab or in the console by any chance?

**Reply by Naveen-Rangarajan** [reply]
Attached are the screen shots.

Post Search Query is a success with status code 200.
![image](https://user-images.githubusercontent.com/61411651/147360408-50af7374-f303-4011-a682-1dd3658fecb5.png)


Network 
![image](https://user-images.githubusercontent.com/61411651/147360180-439ff011-fa4b-4348-b5ad-198d2cb5195e.png)

Console
![image](https://user-images.githubusercontent.com/61411651/147360274-da857d6a-5894-400f-95e5-db6e82ac0e8b.png)

Not sure if above errors are related to ODB.

**Reply by FranckyC** [reply]
I see a 401 here, could be related. Did you try with with [**Search Query Tool**](https://github.com/pnp/PnP-Tools/tree/master/Solutions/SharePoint.Search.QueryTool) with the same query for instance?

**Reply by Naveen-Rangarajan** [reply]
401 error is for GetClientSideComponents. Not able to find much info on this error (Is this related to Search API ?).

Regarding the Search Query Tool - 
(1) Running exe files are prohibited in our environment. Same case for Visual Studio and unable to compile the code to check.

(2) However, I was able to connect to my Client's Test tenant in my personal VM and it is returning the ODB results (I was not able to connect to PROD tenant in my personal VM due to restrictions).

Sharing some screen shots 
![image](https://user-images.githubusercontent.com/61411651/147437408-ac364698-d44b-4e96-833b-f5cb686cbb78.png)

Status 
HTTP/1.1 200 OK
	Serialized Query:
Query Culture="en-US" EnableStemming="True" EnablePhonetic="False" EnableNicknames="False" IgnoreAllNoiseQuery="True" SummaryLength="180" MaxSnippetLength="180" DesiredSnippetLength="90" KeywordInclusion="0" QueryText="MediaLibrary_UserGuide.docx" QueryTemplate="" TrimDuplicates="True" Site="1148342b-08b6-491d-bc9a-8b3e7a86c811" Web="9131de9a-4d70-4042-9f62-dd6a80a0bac1" KeywordType="True" HiddenConstraints="" 

	Elapsed Time (ms): 54

	Primary Query Results:
		Total Rows: 1
		Total Rows Including Duplicates: 1
		Query Id: e39490f6-ec02-4d1b-a36b-f53dd5857514
		Query Rule Id: 00000000-0000-0000-0000-000000000000
		Query Modification: MediaLibrary_UserGuide.docx -ContentClass=urn:content-class:SPSPeople
		
Primary Results
![image](https://user-images.githubusercontent.com/61411651/147437686-fe8b0f80-1f55-4486-8662-ae39ca8a529d.png)

Still unsure where the root cause is.

**Reply by wobba** [reply]
The query from the web part and the query tool should be similar and return the same results. You can use fiddler to grab the requests and compare them.

**Reply by Naveen-Rangarajan** [reply]
The query in Fiddler --> /sites/home/_api/search/postquery  does not provide much info for OneDrive results.

![image](https://user-images.githubusercontent.com/61411651/148035050-92a8204b-e4aa-4734-a7cb-7def0869b078.png)

**Also there are no errors visible in the fiddler**. The 304 code for Display template can be ignored as it does not affect the one drive result outcome.

![image](https://user-images.githubusercontent.com/61411651/148036351-d23cb0ac-f378-46fb-a193-80f82f187ffc.png)

**Reply by Naveen-Rangarajan** [reply]
@wobba , @FranckyC 
What needs to be done further w.r.t to this issue.

**Reply by wobba** [reply]
You should get OneDrive and SharePoint results in the same request. Nothing special should be needed. If you don't get results as expected I assume the issue is also present in the classic search page? And if so you need to log a support case as it's an API issue.
